assets/js/modules/analytics/components/module/ModulePopularPagesWidget/index.js

```diff
@@ -72,20 +72,27 @@ export default function ModulePopularPagesWidget( { Widget, WidgetReportError }
 	const loaded = useSelect( ( select ) => select( MODULES_ANALYTICS ).hasFinishedResolution( 'getReport', [ args ] ) );
 	const error = useSelect( ( select ) => select( MODULES_ANALYTICS ).getErrorForSelector( 'getReport', [ args ] ) );
 
-	if ( error ) {
-		return <WidgetReportError error={ error } />;
+	if ( ! loaded ) {
+		// Return NULL while loading to replicate the behavior of the legacy components.
+		return null;
 	}
 
+	const isZero = isZeroReport( report );
+	const reportIsValid = ! error && ! isZero;
+
 	return (
 		<Widget
 			Header={ Header }
-			Footer={ Footer }
-			noPadding
+			Footer={ reportIsValid ? Footer : undefined }
```

Comment:
Why not always render the `Footer` here, same way as the `Header`? It has the source link only and that shouldn't depend on whether or not the API request is successful IMO.

Reply:
Updated.